src/dotty/tools/dotc/core/Types.scala

```diff
@@ -869,7 +869,13 @@ object Types {
         case SkolemType(tp) =>
           tp.lookupRefined(name)
         case pre: WildcardType =>
-          WildcardType
+          pre.optBounds match {
+            case TypeBounds(lo, hi) if name.isTypeName =>
+              // for term names, forming a Termref with just the name risks losing overloaded instance
+              WildcardType(TypeBounds(NamedType(lo, name), NamedType(hi, name)))
```

Comment:
The comment on https://github.com/dotty-staging/dotty/blob/fix/shuffle/src/dotty/tools/dotc/core/Types.scala#L1423-L1424 is now wrong since the wildcard is now bounded, not unbounded. it'd be nice to update it and maybe also document the handling of wildcard in `lookupRefined` itself.